app/screens/image_preview/image_preview.js

```diff
@@ -73,219 +64,388 @@ export default class ImagePreview extends PureComponent {
     constructor(props) {
         super(props);
 
-        this.zoomableImages = {};
+        props.navigator.setStyle({
+            screenBackgroundColor: '#000',
+        });
 
-        const currentFile = props.files.findIndex((file) => file.id === props.fileId);
-        this.initialPage = currentFile;
+        this.openAnim = new Animated.Value(0);
+        this.headerFooterAnim = new Animated.Value(1);
+        this.documents = [];
 
         this.state = {
-            currentFile,
-            drag: new Animated.ValueXY(),
-            files: props.files,
-            footerOpacity: new Animated.Value(1),
-            pagingEnabled: true,
-            showFileInfo: true,
-            wrapperViewOpacity: new Animated.Value(0),
-            limitOpacity: new Animated.Value(0),
+            index: props.index,
+            origin: props.origin,
+            showDownloader: false,
+            target: props.target,
         };
     }
 
-    componentWillMount() {
-        this.mainViewPanResponder = PanResponder.create({
-            onMoveShouldSetPanResponderCapture: this.mainViewMoveShouldSetPanResponderCapture,
-            onPanResponderMove: Animated.event([null, {
-                dx: 0,
-                dy: this.state.drag.y,
-            }]),
-            onPanResponderRelease: this.mainViewPanResponderRelease,
-            onPanResponderTerminate: this.mainViewPanResponderRelease,
-        });
+    componentDidMount() {
+        this.startOpenAnimation();
     }
 
-    componentDidMount() {
-        InteractionManager.runAfterInteractions(() => {
-            Animated.timing(this.state.wrapperViewOpacity, {
-                toValue: 1,
-                duration: 100,
-            }).start();
-        });
+    componentWillUnmount() {
+        StatusBar.setHidden(false, 'fade');
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (!nextProps.files.length) {
-            this.showDeletedFilesAlert();
-        }
+    get gallery() {
```

Comment:
Can you make these regular methods? They don't make sense as getters

Reply:
will do